ambry-frontend/src/main/java/com.github.ambry.frontend/AmbrySecurityService.java

```diff
@@ -286,18 +283,12 @@ private void setBlobPropertiesHeaders(BlobProperties blobProperties, RestRespons
    */
   private void setAccountAndContainerHeaders(RestRequest restRequest, RestResponseChannel restResponseChannel)
       throws RestServiceException {
-    Object accountObj = restRequest.getArgs().get(RestUtils.InternalKeys.TARGET_ACCOUNT_KEY);
-    Object containerObj = restRequest.getArgs().get(RestUtils.InternalKeys.TARGET_CONTAINER_KEY);
-    if (accountObj != null && accountObj instanceof Account) {
-      Account account = (Account) accountObj;
-      if (account.getId() != Account.UNKNOWN_ACCOUNT_ID) {
-        restResponseChannel.setHeader(RestUtils.Headers.TARGET_ACCOUNT_NAME, ((Account) accountObj).getName());
-        if (containerObj != null && containerObj instanceof Container) {
-          Container container = (Container) containerObj;
-          restResponseChannel.setHeader(RestUtils.Headers.TARGET_CONTAINER_NAME, container.getName());
-          restResponseChannel.setHeader(RestUtils.Headers.PRIVATE, container.isPrivate());
-        }
-      }
+    Account account = RestUtils.getAccountFromArgs(restRequest.getArgs());
+    Container container = RestUtils.getContainerFromArgs(restRequest.getArgs());
+    if (account.getId() != Account.UNKNOWN_ACCOUNT_ID) {
+      restResponseChannel.setHeader(RestUtils.Headers.TARGET_ACCOUNT_NAME, account.getName());
+      restResponseChannel.setHeader(RestUtils.Headers.TARGET_CONTAINER_NAME, container.getName());
     }
+    restResponseChannel.setHeader(RestUtils.Headers.PRIVATE, !container.isCacheable());
```

Comment:
why do we need to set the PRIVATE header ? in other words, why needs  this header?

Reply:
For now, some clients might expect this header to present when fetching blob info. Did not want to change the rest API.